pennylane/devices/tests/test_compare_default_qubit.py

```diff
@@ -0,0 +1,201 @@
+            qml.RX(phi, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(wires=0) @ qml.PauliZ(wires=1))
+
+        qnode_def = qml.QNode(circuit, dev_def)
+        qnode = qml.QNode(circuit, dev)
+
+        grad_def = qml.grad(qnode_def, argnum=[0, 1])
+        grad = qml.grad(qnode, argnum=[0, 1])
+
+        assert np.allclose(qnode(theta, phi), qnode_def(theta, phi), atol=tol(dev.analytic))
+        assert np.allclose(grad(theta, phi), grad_def(theta, phi), atol=tol(dev.analytic))
+
+    @pytest.mark.parametrize("ret", [qml.expval, qml.var])
+    def test_random_circuit(self, device, tol, ret):
+        """Test that the expectation value of a random circuit is correct"""
+        n_wires = 2
+        dev = device(n_wires)
+        dev_def = qml.device("default.qubit", wires=n_wires)
+
+        if dev.name == dev_def.name:
+            pytest.skip("Device is default.qubit.")
+
+        supports_tensor = (
+            "supports_tensor_observables" in dev.capabilities()
+            and dev.capabilities()["supports_tensor_observables"]
+        )
+
+        if not supports_tensor:
+            pytest.skip("Device does not support tensor observables.")
+
+        if not dev.analytic:
+            pytest.skip("Device is in non-analytical mode.")
+
+        n_layers = np.random.randint(1, 5)
+        weights = 2 * np.pi * np.random.rand(n_layers, 1)
+
+        def circuit(weights):
+            RandomLayers(weights, wires=range(n_wires))
+            return ret(qml.PauliZ(wires=0) @ qml.PauliX(wires=1))
+
+        qnode_def = qml.QNode(circuit, dev_def)
+        qnode = qml.QNode(circuit, dev)
+
+        grad_def = qml.grad(qnode_def, argnum=0)
+        grad = qml.grad(qnode, argnum=0)
+
+        assert np.allclose(qnode(weights), qnode_def(weights), atol=tol(dev.analytic))
+        assert np.allclose(grad(weights), grad_def(weights), atol=tol(dev.analytic))
+
+    def test_four_qubit_random_circuit(self, device, tol):
+        """Test a four-qubit random circuit with the whole set of possible gates"""
+        n_wires = 4
+        dev = device(n_wires)
+        dev_def = qml.device("default.qubit", wires=n_wires)
+
+        if dev.name == dev_def.name:
+            pytest.skip("Device is default.qubit.")
+
+        if not dev.analytic:
+            pytest.skip("Device is in non-analytical mode.")
+
+        gates = [
+            qml.PauliX,
+            qml.PauliY,
+            qml.PauliZ,
+            qml.S,
+            qml.T,
+            qml.RX,
+            qml.RY,
+            qml.RZ,
+            qml.Hadamard,
+            qml.Rot,
+            qml.CRot,
+            qml.Toffoli,
+            qml.SWAP,
+            qml.CSWAP,
+            qml.U1,
+            qml.U2,
+            qml.U3,
+            qml.CRX,
+            qml.CRY,
+            qml.CRZ,
+        ]
+
+        layers = 3
+        np.random.seed(1967)
+        gates_per_layers = [np.random.permutation(gates) for _ in range(layers)]
```

Comment:
```suggestion
        gates_per_layers = [np.random.permutation(gates).numpy() for _ in range(layers)]
```
This is needed since we are now importing numpy from PL